### DISCUSSION TITLE
Adding links to the top bar in Filament 3

### DISCUSSION THREAD

**Discussion by AllTheUserNamesAreGone** [filamentphp/filament#12117] [Help]
### Package

Panel builder

### Package Version

v3

### How can we help you?

Hi,
How do you add links to the top nav (header) bar? I've tried things like ->topNavigation(); but that moves all the normal side navigation items to the top. I want the normal side nav where it is plus adding some links or route items to the top so I have both a top and a side navigation. So far all I can do is add items to the top nav dropdown menu on the right hand side. I'm trying to add links along the top. Such as:

Home | Our Services | Blog | About Us | Contact Us etc

Any help much appreciated.
Thanks

**Comment by leandrocfe** [comment]
Use render hooks: 
https://filamentphp.com/docs/3.x/panels/configuration#render-hooks
https://filamentphp.com/docs/3.x/support/render-hooks#available-render-hooks

**Comment by AllTheUserNamesAreGone** [comment]
OK, thanks I'll have a look at that.

**Reply by AllTheUserNamesAreGone** [reply]
Hi,

I have managed to inject a blade view file with the menu within it into the top nav bar. However only a few Tailwind styles are being applied. For example class="border-2 border-green-500" produces a grey border not a green border. If I change border-2 to border-4 the border then  disappears. If I use the hover: it does nothing. Most of the tailwind css doesn't work with the exception of margins and padding. I have Filament installed with Jetstream and was wondering if this was causing a problem?

Any help much appreciated. Thanks.

**Reply by leandrocfe** [reply]
Did you create a custom theme?

https://filamentphp.com/docs/3.x/panels/themes/#creating-a-custom-theme

**Reply by AllTheUserNamesAreGone** [reply]
I did try adding classes to the app.css and did look at creating a custom theme but as even the custom classes were not getting 'through' I didn't think a custom theme would either. 

Everything did work as expected when I added the Tailwind CDN directly to the blade file.
<script src="https://cdn.tailwindcss.com"></script>

Not the best solution but Filament is taking up too much time to do simple little things. I'll try to make time to give a  custom theme a go.

Thanks very much for your help.

**Reply by AllTheUserNamesAreGone** [reply]
I created a custom theme and the topnav theme started to work without having to add anything to the filament/theme.css. I wasn't expecting that to happen and I'm not sure why the theming now works in the same blade even though I haven't added anything, only had left what was there from yesterdays' attempts that didn't work. 

Thanks again for your help.

**Reply by leandrocfe** [reply]
Previously, these classes were not automatically compiled by Tailwind CSS. However, now they are, using the content configuration in your tailwind.config.js file.

**Reply by AllTheUserNamesAreGone** [reply]
Oh I see. I think? Thanks.
